ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -355,30 +353,6 @@ protected void scheduleTasks() {
       cloudCompactionScheduler.scheduleWithFixedDelay(cloudStorageCompactor, cloudConfig.cloudBlobCompactionStartupDelaySecs,
           TimeUnit.HOURS.toSeconds(cloudConfig.cloudBlobCompactionIntervalHours), TimeUnit.SECONDS);
     }
-
-    // Schedule thread to purge blobs belonging to deprecated containers for this VCR's partitions
-    // after delay to allow startup to finish.
-    scheduleTask(() -> cloudContainerCompactor.compactAssignedDeprecatedContainers(
-            vcrClusterParticipant.getAssignedPartitionIds()), cloudConfig.cloudContainerCompactionEnabled,
-        cloudConfig.cloudContainerCompactionStartupDelaySecs,
-        TimeUnit.HOURS.toSeconds(cloudConfig.cloudContainerCompactionIntervalHours), "cloud container compaction");
-  }
-
-  /**
-   * Schedule the specified task if enabled with the specified delay and interval.
-   * @param task {@link Runnable} task to be scheduled.
-   * @param isEnabled flag indicating if the task is enabled. If false the task is not scheduled.
-   * @param delaySec initial delay to allow startup to finish before starting task.
-   * @param intervalSec period between successive executions.
-   * @param taskName name of the task being scheduled.
-   */
-  private void scheduleTask(Runnable task, boolean isEnabled, long delaySec, long intervalSec, String taskName) {
-    if (isEnabled) {
-      scheduler.scheduleAtFixedRate(task, delaySec, intervalSec, TimeUnit.SECONDS);
-      logger.info("Scheduled {} task to run every {} seconds starting in {} seconds.", taskName, intervalSec, delaySec);
-    } else {
-      logger.warn("Running with {} turned off!", taskName);
-    }
```

Comment:
why are we removing the container deprecation logic here. We still deletes containers and once those containers are deleted, we need cloud to clean up all the blobs that belong to those containers.

Reply:
i merged container-deprecation code with regular cloud-compaction. the cloud-compaction scans all blobs and checks if any belongs to a deprecated container and deletes them in its regular flow.

imo, this separate job for deprecated containers was horrible idea.

Reply:
that makes sense.